### PR TITLE
feat(search): Better invalid field state

### PR DESCRIPTION
**Pull request by jennmueng**
Nicer invalid field state with a link to the docs on all searchable properties.

Before:
Screen Shot 2022-06-09 at 3 25 46 PM

After:
Screen Shot 2022-06-10 at 1 02 01 PM
Screen Shot 2022-06-10 at 1 01 51 PM
Screen Shot 2022-06-15 at 11 04 41 AM
Screen Shot 2022-06-15 at 11 05 28 AM

### CODE REVIEW

**Comment by jennmueng** [comment]
@evanpurkhiser Had to add an `await tick()` in the widget builder test here for it to work again

**Comment by jennmueng** [comment]
@evanpurkhiser Had to add an `await tick()` in the widget builder test here for it to work again

File: static/app/components/smartSearchBar/searchDropdown.tsx [hunk 1]
```diff
@@ -31,6 +31,15 @@ class SearchDropdown extends PureComponent<Props> {
   renderDescription = (item: SearchItem) => {
     const searchSubstring = this.props.searchSubstring;
     if (!searchSubstring) {
+      if (item.type === ItemType.INVALID_TAG) {
+        return (
+          <Invalid>
+            The field <strong>{item.desc}</strong> isn't supported here.{' '}
+            <Highlight>See all searchable properties in the docs.</Highlight>
```
Comment (priscilawebdev) [inline static/app/components/smartSearchBar/searchDropdown.tsx:38]:
```suggestion
 {tct(
              "The field [field] isn't supported here. [highlight:See all searchable properties in the docs.]",
              {
                field: <strong>{item.desc}</strong>,
                highlight: <Highlight />,
              }
            )}
```

File: static/app/components/smartSearchBar/searchDropdown.tsx [hunk 2]
```diff
@@ -65,21 +74,23 @@ class SearchDropdown extends PureComponent<Props> {
...
+    return (
+      <SearchListItem
+        key={item.value || item.desc || item.title}
+        className={item.active ? 'active' : undefined}
+        data-test-id="search-autocomplete-item"
+        onClick={item.callback ?? this.props.onClick.bind(this, item.value, item)}
+        ref={element => item.active && element?.scrollIntoView?.({block: 'nearest'})}
+      >
+        <SearchItemTitleWrapper>
+          {item.title && item.title + (item.desc ? ' · ' : '')}
```
Comment (priscilawebdev) [inline static/app/components/smartSearchBar/searchDropdown.tsx:87]: Nit:

```suggestion
  {item.title && `${item.title}${item.desc ? ' · ' : ''}`}
```

File: static/app/components/smartSearchBar/searchDropdown.tsx [hunk 3]
```diff
@@ -289,3 +298,13 @@ const HotkeyGlyphWrapper = styled('span')`
 const HotkeyTitle = styled(`span`)`
   font-size: ${p => p.theme.fontSizeSmall};
 `;
+
+const Invalid = styled(`span`)`
+  font-size: ${p => p.theme.fontSizeSmall};
+  font-family: ${p => p.theme.text.family};
+  color: ${p => p.theme.gray400};
```
Comment (priscilawebdev) [inline static/app/components/smartSearchBar/searchDropdown.tsx:305]: does it look good on the dark mode too? maybe we should use an alias from our theme?

**Comment by priscilawebdev** [comment]
should we break the word in this case instead of using ellipsis?

image

**Comment by priscilawebdev** [comment]
> @evanpurkhiser Had to add an await tick() in the widget builder test here for it to work again

not sure why this was needed

**Comment by jennmueng** [comment]
Thanks @priscilawebdev!

**Review by evanpurkhiser — APPROVED** [review]

File: static/app/components/smartSearchBar/searchDropdown.tsx [hunk 4]
```diff
@@ -65,21 +78,23 @@ class SearchDropdown extends PureComponent<Props> {
...
+        ref={element => item.active && element?.scrollIntoView?.({block: 'nearest'})}
+      >
+        <SearchItemTitleWrapper>
+          {item.title && `${item.title}${item.desc ? ' · ' : ''}`}
+          <Description>{this.renderDescription(item)}</Description>
+          <Documentation>{item.documentation}</Documentation>
+        </SearchItemTitleWrapper>
+      </SearchListItem>
+    );
+  };
```
[the remark this answers is not in the document]
Comment (jennmueng) [reply static/app/components/smartSearchBar/searchDropdown.tsx:97]: Huh good catch, originally was trying out a different implementation and forgot to revert the function back

**Pull request merged**